### PR TITLE
Metatx Popup

### PR DESCRIPTION
**Pull request by Velenir**
Popup for meta transaction

- [x] Extend and reexport state/application/actions
- [x] Change path to absolute where extended exports are used
- [x] Actually make the `<MetaTxPopup/>`
- [x] Use it where applicable
- [x] Fix `lastCheckedBlock` reset on `orders/clearOrders` action

localhost_3000_ (1)

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr73--gpv2swap.review.gnosisdev.com)**: Testing web app

File: src/custom/components/Popups/PopupItem.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    const {
+      listUpdate: { listUrl, oldList, newList, auto }
+    } = content
+    popupContent = <ListUpdatePopup popKey={popKey} listUrl={listUrl} oldList={oldList} newList={newList} auto={auto} />
+  } else if ('metatxn' in content) {
+    const {
+      metatxn: { id, success, summary }
+    } = content
+    popupContent = <MetaTransactionPopup id={id} success={success} summary={summary} />
+  }
```
Comment (Velenir) [inline src/custom/components/Popups/PopupItem.tsx:91]: This is the only part I changed, everything else is copy-:spaghetti:

File: src/custom/hooks/useSwapCallback.ts [hunk 2]
```diff
@@ -85,6 +89,29 @@ export function useSwapCallback(
...
+        const addPendingOrderAndPopup: typeof addPendingOrder = pendingOrderParams => {
+          batch(() => {
+            addPendingOrder(pendingOrderParams)
+
+            const {
+              id,
+              order: { summary }
+            } = pendingOrderParams
+
+            addPopup(
```
Comment (Velenir) [inline src/custom/hooks/useSwapCallback.ts:102]: I still plan on making that Popup middleware, then this code will go away

File: src/custom/components/Popups/MetaTransactionPopup.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { ExternalLink } from 'theme'
+import { AutoColumn } from 'components/Column'
+import { AutoRow } from 'components/Row'
+import { getOrderLink } from 'utils/operator'
+
+const RowNoFlex = styled(AutoRow)`
+  flex-wrap: nowrap;
+`
+
+export default function MetaTransactionPopup({
```
Comment (Velenir) [inline src/custom/components/Popups/MetaTransactionPopup.tsx:15]: Modeled after `<TransactionPopup/>`

File: src/custom/components/Popups/MetaTransactionPopup.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  const theme = useContext(ThemeContext)
+
+  return (
+    <RowNoFlex>
+      <div style={{ paddingRight: 16 }}>
+        {success ? <CheckCircle color={theme.green1} size={24} /> : <AlertCircle color={theme.red1} size={24} />}
+      </div>
+      <AutoColumn gap="8px">
+        <TYPE.body fontWeight={500}>{summary ?? 'ID: ' + id.slice(0, 8) + '...' + id.slice(58, 65)}</TYPE.body>
```
Comment (W3stside) [inline src/custom/components/Popups/MetaTransactionPopup.tsx:34]: can this be a function or sth elsewhere?

File: src/custom/components/Popups/PopupItem.tsx (hunk 1 above)
Comment (W3stside) [reply src/custom/components/Popups/PopupItem.tsx:91]: probably best to create a `PopupItemMod.tsx` and import the styled components

**Comment by W3stside** [comment]
very nice! checked it worked and love the link to api

**Review by W3stside — APPROVED** [review]

File: src/custom/components/Popups/PopupItem.tsx (hunk 1 above)
Comment (Velenir) [reply src/custom/components/Popups/PopupItem.tsx:91]: maybe, but as it is a copypasta component with just a couple lines changed, let's not

File: src/custom/components/Popups/MetaTransactionPopup.tsx (hunk 4 above)
Comment (Velenir) [reply src/custom/components/Popups/MetaTransactionPopup.tsx:34]: It's exact code from `TransactionPopup.tsx`. If we ever need it in more places, then let's make a util

Comment (anxolin) [reply src/custom/components/Popups/MetaTransactionPopup.tsx:34]: agree is a bit ugly those slices...

**Review by anxolin — COMMENTED** [review]
Nice PR! I think is a bit of copy paste done, and maybe there was some ways around. But let's leave it for now

**Pull request merged**